container/container.go

```diff
@@ -505,7 +505,8 @@ func copyEscapable(dst io.Writer, src io.ReadCloser, keys []byte) (written int64
 func (container *Container) ShouldRestart() bool {
 	return container.HostConfig.RestartPolicy.Name == "always" ||
 		(container.HostConfig.RestartPolicy.Name == "unless-stopped" && !container.HasBeenManuallyStopped) ||
-		(container.HostConfig.RestartPolicy.Name == "on-failure" && container.ExitCode != 0)
+		(container.HostConfig.RestartPolicy.Name == "on-failure" && container.ExitCode != 0 &&
```

Comment:
Can we have `container/monitor.go` use this fn as well? It looks like it's defining it's own rules, which are exactly the same.